rotkehlchen/externalapis/etherscan.py

```diff
@@ -236,6 +241,55 @@ def _query(
 
         return result
 
+    @staticmethod
+    def _get_or_create_genesis_transaction(
+            database: DBHandler,
+            account: ChecksumEthAddress,
+    ) -> EthereumTransaction:
+        """Add 0x0 ethereum address to DB if it is missing"""
+        db_ethx = DBEthTx(database)
+
+        # Fetch the genesis txn from DB
+        result = db_ethx.get_ethereum_transactions(
+            filter_=ETHTransactionsFilterQuery.make(
+                tx_hash=deserialize_evm_tx_hash(GENESIS_HASH),
+            ),
+            has_premium=False,
+        )
+        if len(result) == 1:
+            return result[0]
+
+        # Add the 0x0 transaction hash to known transactions
+        transaction = EthereumTransaction(
+            tx_hash=deserialize_evm_tx_hash(GENESIS_HASH),
+            block_number=0,
+            timestamp=Timestamp(1438269973),
+            from_address=deserialize_ethereum_address(GENESIS_HASH),
+            to_address=None,
+            value=0,
+            gas=0,
+            gas_price=0,
+            input_data=b'',
+            gas_used=0,
+            nonce=0,
+        )
+
+        # Add the 0x0 parent txn to DB with relevant_address as
+        # the first address to have transaction in genesis block
+        db_ethx.add_ethereum_transactions(
+            ethereum_transactions=[transaction],
+            relevant_address=account,
+        )
+        return transaction
+
+    @staticmethod
+    def _latest_trace_id(database: 'DBHandler') -> int:
+        cursor = database.conn.cursor()
+        trace_id, = cursor.execute(
+            'SELECT MAX(trace_id) from ethereum_internal_transactions',
+        ).fetchone()
+        return trace_id + 1 if trace_id else 0
+
```

Comment:
I'm not sure on 100% but. Are you sure that doing `trace_id + 1` approach will work reliably? As I mentioned [here](https://github.com/rotki/rotki/pull/4475/files#diff-b0396843e02a7553886d426c1f9ec9ff779e0f122a8fb05547322d18149a8c54R562) you potentially can get conflict with `traceId` returned by etherscan.

Reply:
This was the approach I discussed with Yabir, do you have any other alternative to this?

Reply:
If you are sure that there will be no conflicts with etherscan's tradeIds then should be okay.

Hm, as far as I know etherscan doesn't return `tradeId` for genesis transactions, right? So for this particular transaction should be fine then to use custom `trace_id` indeed

Reply:
Hm, I am also not too sure about that, will have to talk with left.

Reply:
No problem with using custom `trace_id` in my opinion